pkg/virt-controller/services/template.go

```diff
@@ -22,27 +22,52 @@ package services
 import (
 	"path/filepath"
 	"strconv"
+	"strings"
 
-	kubev1 "k8s.io/api/core/v1"
+	k8sv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
 
 	"kubevirt.io/kubevirt/pkg/api/v1"
 	"kubevirt.io/kubevirt/pkg/precond"
-	"kubevirt.io/kubevirt/pkg/registry-disk"
+	registrydisk "kubevirt.io/kubevirt/pkg/registry-disk"
 )
 
+const configMapName = "kube-system/virt-controller"
+const allowEmulationKey = "debug.allowEmulation"
+
 type TemplateService interface {
-	RenderLaunchManifest(*v1.VirtualMachine) *kubev1.Pod
+	RenderLaunchManifest(*v1.VirtualMachine) (*k8sv1.Pod, error)
 }
 
 type templateService struct {
 	launcherImage   string
 	virtShareDir    string
 	imagePullSecret string
+	store           cache.Store
+}
+
+func IsEmulationAllowed(store cache.Store) (bool, error) {
+	obj, exists, err := store.GetByKey(configMapName)
+	if err != nil {
+		return false, err
+	}
+	if !exists {
+		return exists, nil
+	}
+	allowEmulation := false
+	cm := obj.(*k8sv1.ConfigMap)
+	emu, ok := cm.Data[allowEmulationKey]
+	if ok {
+		// TODO: is this too specific? should we just look for the existence of
+		// the 'allowEmulation' key itself regardless of content?
```

Comment:
I think we should look for 'true' like you're doing.  It follows the principle of least surprise.

For example, if we looked just to see if the key was available to enable emulation, and the value was set to 'false', i'd be kind of surprised to discover emulation was allowed.